### ISSUE TITLE
Bug: no-misused-promises/checksVoidReturn confuses static methods with non-static ones

### ISSUE THREAD

**Issue by adrien-febvay** [typescript-eslint/typescript-eslint#9942]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22package%3A+eslint-plugin%22) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Playground Link

https://typescript-eslint.io/play/#ts=5.5.2&showAST=es&fileType=.tsx&code=LAKAxgNghgzjAEAhWBTeBvU94AcCuARhAJZjwwD2AtigLIoAuAFhQCYAUAlAFzwBuFYqwzwAvqHEhQkWAgCiADwYoAdqxTCUS1awTIYaTCGz4ipcgygNzsAJ4qylGvWZsuvAAoAnasQMAeASEAPgxJSSA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1oFtLlZkiACa1i0Dr0GoMkRNHHRI4MAF8QKoA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
class Base {
  public someMethod() {}
}

class Extended extends Base {
  public static async someMethod() {}
}
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/no-misused-promises": "error"
  },
};
```

### Expected Result

I expected to have no error issued.

### Actual Result

The following error was issued:
```
Promise-returning method provided where a void return was expected by extended/implemented type 'Base'. 6:3 - 6:53
```

### Additional Info

When a class declares a static method which has the same name as a non-static method in the extended class, the `no-misused-promises/checksVoidReturn` rule mistakenly consider the static method extends the non-static one. As a result, if the static method returns a `Promise`, but the non-static method returns `void`, an error is issue.

Of course, the two methods, while having the same name, are completely distinct and shouldn't be compared in the first place.
